workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestTemplate.cs

```diff
@@ -0,0 +1,335 @@
+
+            componentInterest.Queries.Clear();
+            componentInterest.Queries.AddRange(componentInterestQueries);
+            return this;
+        }
+
+        /// <summary>
+        ///     Add interestQueries to the Interest component.
+        /// </summary>
+        /// <typeparam name="T">
+        ///     Type of the authoritative component to clear interestQueries from.
+        /// </typeparam>
+        /// <returns>
+        ///     An InterestTemplate object.
+        /// </returns>
+        public InterestTemplate ClearQueries<T>()
+            where T : ISpatialComponentData
+        {
+            return ClearQueries(Dynamic.GetComponentId<T>());
+        }
+
+        /// <summary>
+        ///     Add interestQueries to the Interest component.
+        /// </summary>
+        /// <param name="componentId">
+        ///     Component ID of the authoritative component to clear interestQueries from.
+        /// </param>
+        /// <returns>
+        ///     An InterestTemplate object.
+        /// </returns>
+        public InterestTemplate ClearQueries(uint componentId)
+        {
+            interest.Remove(componentId);
+            return this;
+        }
+
+        /// <summary>
+        ///     Removes all interestQueries.
+        /// </summary>
+        /// <returns>
+        ///     An InterestTemplate object.
+        /// </returns>
+        public InterestTemplate ClearAllQueries()
+        {
+            interest.Clear();
+            return this;
+        }
+
+        /// <summary>
+        ///     Builds the Interest snapshot.
+        /// </summary>
+        /// <returns>
+        ///     A Interest.Snapshot object.
+        /// </returns>
+        public Interest.Snapshot ToSnapshot()
+        {
+            return new Interest.Snapshot { ComponentInterest = interest };
+        }
+
+        /// <summary>
+        ///     Returns a deep copy of the underlying data of an Interest component.
+        /// </summary>
+        /// <returns>
+        ///     A Dictionary<uint, ComponentInterest>.
+        /// </returns>
+        public Dictionary<uint, ComponentInterest> ToComponentInterest()
+        {
+            return DeepCopy(interest);
+        }
+
+        /// <summary>
+        ///     Returns the underlying data of an Interest component.
+        /// </summary>
+        /// <returns>
+        ///     A Dictionary<uint, ComponentInterest>.
+        /// </returns>
+        public Dictionary<uint, ComponentInterest> AsComponentInterest()
+        {
+            return interest;
+        }
+
+        private static List<ComponentInterest.Query> QueryParamsToList(InterestQuery interestQuery,
+            params InterestQuery[] interestQueries)
+        {
+            var output = new List<ComponentInterest.Query>(interestQueries.Length + 1)
+            {
+                interestQuery.AsComponentInterestQuery()
+            };
+
+            foreach (var constraintParam in interestQueries)
```

Comment:
Can `AddRange`

Reply:
same as here: https://github.com/spatialos/gdk-for-unity/pull/654#discussion_r271237927